### PR TITLE
fix #20614 Add __traits(hasMoveConstructor, type)

### LINKED ISSUE #20614: Add __traits(hasMoveConstructor, type)
As a complement to __traits(hasCopyConstructor, type)

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

**Review by thewilsonator — CHANGES_REQUESTED** [review]
Looks good to me. Needs a Spec PR

**Comment by WalterBright** [comment]
https://github.com/dlang/dmd/issues/20614

**Comment by WalterBright** [comment]
spec pr: https://github.com/dlang/dlang.org/pull/4152

**Review by thewilsonator — APPROVED** [review]

File: compiler/src/dmd/id.d [hunk 1]
```diff
@@ -519,6 +519,7 @@ immutable Msgtable[] msgtable =
     { "getLocation" },
     { "hasPostblit" },
     { "hasCopyConstructor" },
+    { "hasMoveConstructor" },
```
Comment (thewilsonator) [inline compiler/src/dmd/id.d:522]:
```diff
--- a/home/circleci/dmd/compiler/src/dmd/frontend.h
+++ b/home/circleci/dmd/generated/linux/release/64/frontend.h
@@ -8978,6 +8978,7 @@ struct Id final
     static Identifier* getLocation;
     static Identifier* hasPostblit;
     static Identifier* hasCopyConstructor;
+    static Identifier* hasMoveConstructor;
     static Identifier* isCopyable;
     static Identifier* toType;
     static Identifier* parameters;
```
Need to update `frontend.h`

Comment (WalterBright) [reply compiler/src/dmd/id.d:522]: done

**Pull request merged**